src/gnark_backend_glue/mod.rs

```diff
@@ -28,10 +32,81 @@ cfg_if::cfg_if! {
     }
 }
 
-pub fn prove(_circuit: Circuit, _values: Vec<FieldElement>) -> Result<Vec<u8>> {
-    todo!()
+extern "C" {
+    fn Verify(rawr1cs: GoString, proof: GoString) -> c_uchar;
+    fn Prove(rawr1cs: GoString) -> *const c_char;
+}
+
+#[derive(Debug)]
+#[repr(C)]
+struct GoString {
+    ptr: *const c_char,
+    length: i64,
+}
+
+impl GoString {
+    pub fn from_cstring(c_str: &CString) -> GoString {
+        let ptr = c_str.as_ptr();
+        GoString {
+            ptr,
+            length: c_str.as_bytes().len() as i64,
+        }
```

Comment:
`as` casts may panic, we should avoid them as we do with the `unwraps`, you could use something like this instead:
```suggestion
    pub fn from_cstring(c_str: &CString) -> GoString {
        let ptr = c_str.as_ptr();
        let length: i64 = c_str.as_bytes().len().try_into().unwrap();
        GoString { ptr, length }
```

If we want to remove the `unwrap`, the function should return a result like this:
```suggestion
    pub fn from_cstring(c_str: &CString) -> Result<GoString> {
        let ptr = c_str.as_ptr();
        let length: i64: c_str.as_bytes().len().try_into()?
        Ok(GoString { ptr, length })
```

Reply:
Or we could replace the `i64` with an `usize`. It is pretty common to use `usize` to represent the length of something.